src/main/java/com/faforever/client/vault/VaultFileSystemLocationChecker.java

```diff
@@ -0,0 +1,62 @@
+package com.faforever.client.vault;
+
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.Action;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.PersistentNotification;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.preferences.ForgedAlliancePrefs;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.base.CharMatcher;
+import com.sun.security.auth.module.NTSystem;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+import static org.bridj.Platform.isUnix;
+import static org.bridj.Platform.isWindows;
+
+/**
+ * Checks if the user is on linux, has a none ascii username or has One Drive installed. In this cases we need to use a fallback location for the
+ * vault.
+ */
+@Component
+public class VaultFileSystemLocationChecker {
+  public static final Pattern ONE_DRIVE_FOLDER_NAME_IDENTIFIER = Pattern.compile("[Oo]ne[ ]*[Dd]rive");
+  private final NotificationService notificationService;
+  private final PreferencesService preferencesService;
+  private final I18n i18n;
+
+  public VaultFileSystemLocationChecker(NotificationService notificationService, PreferencesService preferencesService, I18n i18n) {
+    this.notificationService = notificationService;
+    this.preferencesService = preferencesService;
+    this.i18n = i18n;
+  }
+
+  public void checkVaultFileSystemLocation() {
+    String userName = new NTSystem().getName();
+    ForgedAlliancePrefs forgedAlliance = preferencesService.getPreferences().getForgedAlliance();
+    if (forgedAlliance.isVaultCheckDone()) {
+      return;
+    }
+    Action revertAction = new Action(i18n.get("revert.alreadySymlinked"), event -> forgedAlliance.setVaultBaseDirectory(preferencesService.getPrimaryVaultLocation()));
+    if (userName != null && !CharMatcher.ascii().matchesAllOf(userName) && isWindows()) {
```

Comment:
We're doing almost the same thing in all these branches. We could put logic in an enum that returns default FA directory status (OK, Non-ascii username, OneDrive or unix) and do base dir setup and notification in one place.

Reply:
Did not use an enum but used extracted to a Method